wfl/Controllers/RestaurantsController.cs

```diff
@@ -25,7 +26,7 @@ public class RestaurantsController : ApiController
         [HttpPut]
         public async Task<IHttpActionResult> UpdateAsync([FromUri] string id, [FromBody] RestaurantDto restaurant)
         {
-            if (string.IsNullOrEmpty(id) || string.IsNullOrEmpty(restaurant?.Name))
+            if (string.IsNullOrEmpty(id) || string.IsNullOrEmpty(restaurant?.Name) || (restaurant.MenuLink != null && !Uri.IsWellFormedUriString(restaurant.MenuLink.ToString(), UriKind.Absolute)))
```

Comment:
I think this is good for now, but I think we're also starting to see how handling validation in this manner starts to get verbose and repetitive across methods.  Perhaps there are some strategies we can think of for future enhancement that may reduce the duplication and verboseness.  I have some thoughts we can discuss further at another time.